docs/src/examples/lcs/lcs.jl

```diff
@@ -0,0 +1,305 @@
+x = rand("ATCG", 4096);
+y = rand("ATCG", 8192);
+seq = LCS(x, y)
+length(seq)
+
+# We can also measure the elapsed time for this implementation, which will serve
+# as a base line to assess the performance of other implementations described
+# below.
+
+using BenchmarkTools
+BenchmarkTools.DEFAULT_PARAMETERS.seconds = 1
+t_seq = @belapsed LCS($x, $y)
+
+
+# ### Tiled sequential version
+#
+# We'll now build a tiled version of the same algorithm. The
+# [`TiledIteration.jl`](https://github.com/JuliaArrays/TiledIteration.jl)
+# package implements various tools allowing to define and iterate over disjoint
+# tiles of a larger array. Among these, the `SplitAxis` function allows
+# splitting a range of indices into a given number of chunks:
+
+using TiledIteration
+SplitAxis(1:20, 3)  # split the range 1:20 into 3 chunks of approximately equal sizes
+
+# A tiled version of the previous algorithm is then as simple as filling the
+# chunks one after the other:
+
+function LCS_tiled(x, y, nx, ny)
+    L = Matrix{Int}(undef, 1+length(x), 1+length(y))
+    init_lengths!(L, x, y)
+
+    for irange in SplitAxis(eachindex(x), nx)
+        for jrange in SplitAxis(eachindex(y), ny)
+            fill_lengths!(L, x, y, irange, jrange)
+        end
+    end
+
+    backtrack(L, x, y)
+end
+
+# Here we split the problem into $15 \times 15$ blocks, and check that the tiled
+# version gives the same results as the plain implementation above. Even without
+# parallelization, tiling can already be beneficial in terms of performance
+# because it is more cache-friendly:
+
+nx = ny = 15
+tiled = LCS_tiled(x, y, nx, ny)
+@assert seq == tiled
+
+t_tiled = @belapsed LCS_tiled($x, $y, nx, ny)
+
+# ## Tiled parallel version
+#
+# Parallelizing the tiled version using `DataFlowTasks` is now very
+# straightforward: it only requires annotating the code to expose data
+# dependencies.
+#
+# In our case:
+#
+# - the initialization task writes to the whole `L` array;
+#
+# - filling a tile involves reading `L` for the provided ranges of indices, and
+#   writing to a range of indices shifted by 1;
+#
+# - backtracking reads the whole `L` array.
+#
+# Note that the backtracking task needs to be fetched in order to get the result
+# in a synchronous way and perform an apple-to-apple comparison to the previous
+# implementations.
+
+using DataFlowTasks
+
+function LCS_par(x, y, nx, ny)
+    L = Matrix{Int}(undef, 1+length(x), 1+length(y))
+    DataFlowTasks.@spawn init_lengths!(@W(L), x, y) label="init"
+
+    for (kx, irange) in enumerate(SplitAxis(eachindex(x), nx))
+        for (ky, jrange) in enumerate(SplitAxis(eachindex(y), ny))
+            DataFlowTasks.@spawn begin
+                @R view(L, irange, jrange)
+                @W view(L, irange .+ 1, jrange .+ 1)
+                fill_lengths!(L, x, y, irange, jrange)
+            end label="tile ($kx, $ky)"
+        end
+    end
+
+    bt = DataFlowTasks.@spawn backtrack(@R(L), x, y) label="backtrack"
+    fetch(bt)
+end
```

Comment:
```suggestion
function LCS_par(x, y, nx, ny)
    L = Matrix{Int}(undef, 1+length(x), 1+length(y))
    for (kx, irange) in enumerate(SplitAxis(eachindex(x), nx))
        DataFlowTasks.@spawn begin
            @W(view(L, irange, 1)) .= 0
        end label="init x ($kx,1)"
        for (ky, jrange) in enumerate(SplitAxis(eachindex(y), ny))
            if kx == 1
                DataFlowTasks.@spawn begin
                    @W(view(L, 1, jrange)) .= 0
                end label="init y ($ky)"
            end
            DataFlowTasks.@spawn begin
                @R view(L, irange, jrange)
                @W view(L, irange .+ 1, jrange .+ 1)
                fill_lengths!(L, x, y, irange, jrange)
            end label="tile ($kx, $ky)"
        end
    end

    bt = DataFlowTasks.@spawn backtrack(@R(L), x, y) label="backtrack"
    fetch(bt)
end
```

Reply:
I think this is correct, and it should be better than having a single
`init` block, and have everyone wait for that to start.

Reply:
Strange: I was positively sure I had tested this, and come to the conclusion that it didn't improve things. In my recollection, the first array write triggered the entire array allocation, which took more time than actually iterating over the first row + column.

But I re-tried it right now following your suggestion, and now it seems to work! Did I dream?